### PR TITLE
Refactoring Branch Search

### PR DESCRIPTION
**Pull request by sojanpr**
Moving search and recommendation interfaces to BranchSearchSDK.

Core SDK need only interfaces to list or delete contents form local
search

@aaustin @EvangelosG

### CODE REVIEW

File: Branch-SDK/src/io/branch/search/BranchListContentConnection.java [hunk 1]
```diff
@@ -8,42 +8,31 @@
...
         if (connection_ == null) {
-            connection_ = new BranchSearchServiceConnection();
+            connection_ = new BranchListContentConnection();
         }
         return connection_;
     }
 
     @Override
     public void onServiceConnected(ComponentName name, IBinder service) {
         Log.d("Bridge_test", "onServiceConnected()");
```
Comment (aaustin) [inline Branch-SDK/src/io/branch/search/BranchListContentConnection.java]: Can you make sure to go clean up all the logs if there are any more? Or put them behind our debug flag.

Comment (aaustin) [inline Branch-SDK/src/io/branch/search/BranchListContentConnection.java]: If you do put them behind a debug flag, we need more details in the log message than just the function name.

**Comment by sojanpr** [comment]
Added the fixes/ changes for addressing the review comments

**Comment by aaustin** [comment]
nice 👍

**Pull request merged**